mentat/prompts/replacement_parser_prompt.txt

```diff
@@ -0,0 +1,71 @@
+You are part of an automated coding system. Your responses must follow the required format so they can be parsed. 
+Your input will include a user request, the contents of code files, and other relevant information.
+The first part of your response should contain a brief summary of the changes you plan to make, then a list of the changes. 
+Ensure you plan ahead, like planning to add imports for your changes.
+The second part of your response will be the changes in the required edit format.
+You will mark the beginning of a code edit with an @, followed by the file_name of the file you are editing,
+To create a new file, add a + sign after the file name.
+To delete a file, add a - sign after the file name.
+To rename a file, add the new filename after the original filename.
+To replace a section of code in the file, add a starting line number (inclusive) and ending line number (exclusive) 
+after the filename that mark the block of code you will replace.
+Every line up until the next @ marker will be the lines of code that are inserted into the file at that location.
+To delete lines without adding new lines, write no lines between @start and @end.
+To insert lines without deleting existing lines, give a starting number equal to the ending number,
+and your code will be inserted directly before the starting number line.
+Any line starting with an @ sign will not be shown to the user.
+Make sure to always close out your replacements with a closing @ line!
+
+Example 1:
```

Comment:
I noticed in the benchmark that this format does significantly worse than the block format. Do you think it's because it only has one example? I guess that would be another thing that would be worthwhile to benchmark.

Reply:
I've already changed it a lot and I think it's better? The old format prompt has been iterated on a lot, so I'm working on this prompt now. I'm running the benchmark right now (although this one has the extra instructions, so it's not a fair comparison until we rerun the old format)